test/arcade/core/util/SolverTest.java

```diff
@@ -0,0 +1,226 @@
+package arcade.core.util;
+
+import org.junit.jupiter.api.Test;
+import arcade.core.util.Solver.Equations;
+import arcade.core.util.Solver.Function;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class SolverTest {
+    @Test
+    public void testConstructor_called_returnsException() {
+        assertThrows(
+                UnsupportedOperationException.class,
+                () -> {
+                    new Solver();
+                });
+    }
+
+    @Test
+    public void testEuler_calledWithEquations_returnsAnswer() {
```

Comment:
Can simplify all the `Equations` constructors to:

```java
Equations e = (t, y) -> {
    ...
};
```

Reply:
maybe I'm misunderstanding but the interface for equations requires a dydt method